staging/src/k8s.io/apimachinery/pkg/util/net/http.go

```diff
@@ -132,13 +133,56 @@ func SetTransportDefaults(t *http.Transport) *http.Transport {
 	if s := os.Getenv("DISABLE_HTTP2"); len(s) > 0 {
 		klog.Infof("HTTP2 has been explicitly disabled")
 	} else if allowsHTTP2(t) {
-		if err := http2.ConfigureTransport(t); err != nil {
+		if err := configureHTTP2Transport(t); err != nil {
 			klog.Warningf("Transport failed http2 configuration: %v", err)
 		}
 	}
 	return t
 }
 
+func readIdleTimeoutSeconds() int {
+	ret := 0
+	// User can set the readIdleTimeout to 0 to disable the HTTP/2
+	// connection health check.
+	if s := os.Getenv("HTTP2_READ_IDLE_TIMEOUT_SECONDS"); len(s) > 0 {
```

Comment:
How will we document this? We can only do this if we expect that basically no one will use it, which I hope is the case.

Reply:
> if we expect that basically no one will use it.

That's my assumption. Also this is similar to the [DISABLE_HTTP2](https://github.com/kubernetes/kubernetes/blob/43edb4063262751bbb6ba18daf3fe9582ae3b2f9/staging/src/k8s.io/apimachinery/pkg/util/net/http.go#L132) env var.